File: ALODWebUtility/Common/PEPPDisposition.vb
Imports ALOD.Core.Domain.Lookup
Imports ALOD.Data

Namespace Common

    Public Class PEPPDisposition
        Inherits IntValuedLookupItem

        Protected _active As Integer = 0            ' active

#Region "Properties"

        Public Property Active() As Integer
            Get
                Return _active
            End Get

            Set(value As Integer)
                _active = value
            End Set
        End Property

#End Region

        ' Get all PEPPDisposition records from the lookup table for the Sys Admin
        Public Overrides Function GetRecords() As DataTable
            Dim lkup As ILookupDao = New NHibernateDaoFactory().GetLookupDao()
            Return lkup.GetPEPPDispositions(0, 0).Tables(0)
        End Function

        ' Insert new PEPPDisposition into the lookup table
        Public Sub InsertPEPPDisposition(ByVal Value As Integer, ByVal Name As String, ByVal Active As Integer)
            Dim adapter As New SqlDataStore()
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPDisposition", Value, Name, Active)
        End Sub

        ' Edit/Update an existing PEPPDisposition in the lookup table
        Public Sub UpdatePEPPDisposition(ByVal Value As Integer, ByVal Name As String, ByVal Active As Integer)
            Dim adapter As New SqlDataStore()
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPDisposition", Value, Name, Active)
        End Sub

    End Class

End Namespace

File: ALODWebUtility.CS/Common/PEPPDisposition.cs
using System;
using System.Data;
using ALOD.Core.Domain.Lookup;
using ALOD.Data;

namespace ALODWebUtility.Common
{
    public class PEPPDisposition : IntValuedLookupItem
    {
        protected int _active = 0;            // active

        #region Properties

        public int Active
        {
            get { return _active; }
            set { _active = value; }
        }

        #endregion

        // Get all PEPPDisposition records from the lookup table for the Sys Admin
        public override DataTable GetRecords()
        {
            ILookupDao lkup = new NHibernateDaoFactory().GetLookupDao();
            return lkup.GetPEPPDispositions(0, 0).Tables[0];
        }

        // Insert new PEPPDisposition into the lookup table
        public void InsertPEPPDisposition(int Value, string Name, int Active)
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPDisposition", Value, Name, Active);
        }

        // Edit/Update an existing PEPPDisposition in the lookup table
        public void UpdatePEPPDisposition(int Value, string Name, int Active)
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPDisposition", Value, Name, Active);
        }
    }
}
